packages/admin/admin/src/appHeader/button/AppHeaderButton.styles.tsx

```diff
@@ -1,43 +1,78 @@
-import { ButtonBaseClassKey } from "@mui/material";
-import { Theme } from "@mui/material/styles";
-import { createStyles } from "@mui/styles";
-
-import { AppHeaderButtonProps } from "./AppHeaderButton";
-
-export type AppHeaderButtonClassKey = ButtonBaseClassKey | "inner" | "startIcon" | "endIcon" | "typography";
-
-export const styles = ({ spacing }: Theme) => {
-    return createStyles<AppHeaderButtonClassKey, AppHeaderButtonProps>({
-        root: {
-            height: "100%",
-            borderLeft: "1px solid rgba(255, 255, 255, 0.2)",
-        },
-        inner: {
-            display: "flex",
-            alignItems: "center",
-            justifyContent: "center",
-            height: "100%",
-            minWidth: "var(--header-height)",
-            boxSizing: "border-box",
-            paddingLeft: spacing(4),
-            paddingRight: spacing(4),
-        },
-        disabled: {},
-        focusVisible: {},
-        startIcon: {
-            display: "flex",
-            alignItems: "center",
-            "&:not(:last-child)": {
-                marginRight: spacing(2),
-            },
-        },
-        endIcon: {
-            display: "flex",
-            alignItems: "center",
-            "&:not(:first-child)": {
-                marginLeft: spacing(2),
-            },
-        },
-        typography: {},
-    });
-};
+import { ButtonBase, Typography } from "@mui/material";
+import { css, styled } from "@mui/material/styles";
+
+export type AppHeaderButtonClassKey = "root" | "inner" | "startIcon" | "endIcon" | "typography";
+
+export const Root = styled(ButtonBase, {
+    name: "CometAdminAppHeaderButton",
+    slot: "root",
+    overridesResolver(_, styles) {
+        return [styles.root];
+    },
+})(
+    css`
+        height: 100%;
+        border-left: 1px solid rgba(255, 255, 255, 0.2);
+    `,
+);
+
+export const Inner = styled("div", {
+    name: "CometAdminAppHeaderButton",
+    slot: "inner",
+    overridesResolver(_, styles) {
+        return [styles.inner];
+    },
+})(
```

Comment:
Could we find a better name for this?

Reply:
How about `ContentContainer`? However, the ClassKey has "inner", why rename the Component? Would that not lead to unnecessary complexity when reading code?

Reply:
> That would be an additional breaking change.

Isn't the refactor a breaking change already? I wouldn't mind another breaking change in favor of a better API

Reply:
We could also make it compatible by supporting the new and the old name in the overrides

Reply:
You're right, an additional breaking change probably won't matter. 
Supporting the old version is not necessary imo. 

Maybe `content` would be a better name? 🤔

Reply:
I'll refactor `inner` then 👍